### PR TITLE
Create analyzer interface (...again!)

### PR DESCRIPTION
**Pull request by ryaanahmed**
This PR revives the work closed in https://github.com/dhmit/gender_analysis/pull/158 and is work towards creating a better, common user interface around all analytic methods in this package (https://github.com/dhmit/gender_analysis/issues/157).

Major changes:
- We implement a new base class for analysis classes: `CorpusAnalyzer`. This class manages the initialization of a Corpus and will eventually be home to the most common textual analysis methods. (n.b. the file is called `base_analyzers` to leave room for `DocumentAnalyzer` if it turns out we need that.)
- `GenderProximityAnalyzer` is refactored as a `CorpusAnalyzer` subclass. In doing so, we lose a bit of functionality—the analyzer class used to be able to take in `Union[Document, Corpus, Sequence[Document]]` for texts, but now it's less flexible. I think the right place for this flexibility is in the Corpus init, so I'll open a separate PR allowing this behavior over there.

Minor:
- Moved `SWORDS_ENG` from `gender_analysis.gender.common` to `gender_analysis.text.common`.
- There was some small work in https://github.com/dhmit/gender_analysis/pull/158 in `Document` replacing `name` with `label`, so that's here, too.

### CODE REVIEW

File: gender_analysis/analysis/proximity.py [hunk 1]
```diff
@@ -298,17 +302,17 @@ class GenderProximityAnalyzer:
 
     Instance methods:
         by_date()
-        by_document()
+        by_document(
```
Comment (MBJean) [inline gender_analysis/analysis/proximity.py]: Super minor point: looks like a closing `)` was deleted here accidentally.

**Review by MBJean — APPROVED** [review]
This LGTM! I found one inconsequential docstring update to make, but otherwise this looks to be in line with the direction we've outlined in #157.

File: gender_analysis/analysis/proximity.py (hunk 1 above)
Comment (ryaanahmed) [reply gender_analysis/analysis/proximity.py]: whoooops! fixed.

**Comment by ryaanahmed** [comment]
I just realized we're not getting CI runs on these PRs to `summer-urop-21-refactor`. Updating the CI.yml temporarily here to include this branch.

**Pull request merged**